#2015-05-01  Yunong Jiang  <yunong@lnxsrv04.seas.ucla.edu>
#      * timespec.h (timespec_cmp): Changed the type of diff from int to long
#        int to prevent overflow.
--- coreutils-with-bug/lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ coreutils-fixed/lib/timespec.h	2015-05-01 17:56:41.960710000 -0700
@@ -45,8 +45,12 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  long int diff = a.tv_sec - b.tv_sec;
+  int n;
+  if (diff > 0) n = 1;
+  else if (diff < 0) n = -1;
+  else n = 0;
+  return n ? n : a.tv_nsec - b.tv_nsec;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
